src/asset/asset-registry.js

```diff
@@ -447,17 +447,19 @@ Object.assign(pc, function () {
          * @param {string} url - The url to load.
          * @param {string} type - The type of asset to load.
          * @param {pc.callbacks.LoadAsset} callback - Function called when asset is loaded, passed (err, asset), where err is null if no errors were encountered.
+         * @param {string} [filename] - Optional asset filename.
          * @example
          * app.assets.loadFromUrl("../path/to/texture.jpg", "texture", function (err, asset) {
          *     var texture = asset.resource;
          * });
          */
-        loadFromUrl: function (url, type, callback) {
+        loadFromUrl: function (url, type, callback, filename) {
```

Comment:
I think this is might be polluting the API unnecessarily. Callbacks are expected to be the last argument and since it would be a breaking change to move filename before callback it's not really viable either. How about adding a new method in the `pc.Asset` class for setting/changing the filename instead?

Reply:
Not sure I follow, Asset already exposes name and filename? This was added so the GlbParser can identify whether the data is GLB or GLTF when loading from a blob.